pz_be_services/routers/v1/chat_router.py

```
@@ -0,0 +1,135 @@
+            f"Successfully returned {len(chats)} private chats for user {current_user.get('username')}"
+        )
+        return response
+
+    except HTTPException as e:
+        raise e
+    except Exception as e:
+        logger.exception(f"Error retrieving private chats: {str(e)}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal server error while retrieving private chats",
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)